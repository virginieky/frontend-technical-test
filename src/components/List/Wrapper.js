import styled from 'styled-components';

const Wrapper = styled.ul`
  height: calc(100vh - 130px);
  overflow: auto;
  padding-left: 0;
  margin-bottom: 0;
  list-style-type: none;
`;

export default Wrapper;
