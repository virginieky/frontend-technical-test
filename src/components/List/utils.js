export const sortConversations = (conversations) => {
  return [...conversations].sort((x, y) => {
    return y.lastMessageTimestamp - x.lastMessageTimestamp;
  });
};
