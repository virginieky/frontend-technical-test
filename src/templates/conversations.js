const conversations = [
  {
    id: 1,
    recipientId: 2,
    recipientNickname: 'Jeremie',
    senderId: 1,
    senderNickname: 'Thibaut',
    lastMessageTimestamp: 1625637849,
  },
  {
    id: 2,
    recipientId: 3,
    recipientNickname: 'Patrick',
    senderId: 1,
    senderNickname: 'Thibaut',
    lastMessageTimestamp: 1620284667,
  },
  {
    id: 3,
    recipientId: 1,
    recipientNickname: 'Thibaut',
    senderId: 4,
    senderNickname: 'Elodie',
    lastMessageTimestamp: 1625648667,
  },
  {
    recipientId: 5,
    recipientNickname: 'Maeko',
    senderId: 1,
    senderNickname: 'Thibaut',
    lastMessageTimestamp: 1632230669743,
    id: 4,
  },
  {
    recipientId: 5,
    recipientNickname: 'Maeko',
    senderId: 1,
    senderNickname: 'Thibaut',
    lastMessageTimestamp: 1632230756751,
    id: 5,
  },
  {
    recipientId: 6,
    recipientNickname: 'Duarte',
    senderId: 1,
    senderNickname: 'Thibaut',
    lastMessageTimestamp: 1632230824677,
    id: 6,
  },
];

export default conversations;
