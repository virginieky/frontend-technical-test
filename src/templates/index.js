export { default as conversations } from './conversations';
export { default as messages } from './messages';
export { default as users } from './users';
