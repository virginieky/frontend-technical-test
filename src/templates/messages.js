const messages = [
  {
    id: 1,
    conversationId: 1,
    timestamp: 1625637849,
    authorId: 1,
    body: "Bonjour c'est le premier message de la première conversation",
  },
  {
    id: 2,
    conversationId: 1,
    timestamp: 1625637867,
    authorId: 1,
    body: "Bonjour c'est le second message de la première conversation",
  },
  {
    id: 3,
    conversationId: 1,
    timestamp: 1625648667,
    authorId: 2,
    body: "Bonjour c'est le troisième message de la première conversation",
  },
  {
    id: 4,
    conversationId: 2,
    timestamp: 1620284667,
    authorId: 2,
    body: "Bonjour c'est le premier message de la seconde conversation",
  },
];

export default messages;
