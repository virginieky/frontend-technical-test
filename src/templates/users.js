const users = [
  {
    id: 1,
    nickname: 'Thibaut',
    token: 'xxxx',
  },
  {
    id: 2,
    nickname: 'Jeremie',
    token: 'xxxx',
  },
  {
    id: 3,
    nickname: 'Patrick',
    token: 'xxxx',
  },
  {
    id: 4,
    nickname: 'Elodie',
    token: 'xxxx',
  },
  {
    id: 5,
    nickname: 'Maeko',
    token: 'xxxx',
  },
  {
    id: 6,
    nickname: 'Duarte',
    token: 'xxxx',
  },
];

export default users;
