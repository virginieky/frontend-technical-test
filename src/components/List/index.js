import React from 'react';
import PropTypes from 'prop-types';

import ListCell from '../ListCell';
import Wrapper from './Wrapper';
import { sortConversations } from './utils';

const List = ({ conversations, onCellClick, selectedItem }) => {
  const sortedConversations = sortConversations(conversations);

  return (
    <Wrapper>
      {sortedConversations.map((conversation) => {
        const { id } = conversation;

        return (
          <ListCell
            key={id}
            {...conversation}
            onClick={onCellClick}
            isSelected={id === selectedItem}
          />
        );
      })}
    </Wrapper>
  );
};

List.defaultProps = {
  conversations: [],
  onCellClick: () => {},
  selectedItem: null,
};

List.propTypes = {
  conversations: PropTypes.array,
  onCellClick: PropTypes.func,
  selectedItem: PropTypes.number,
};

export default List;
