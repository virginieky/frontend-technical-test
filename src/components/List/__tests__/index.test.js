import React from 'react';
import { render } from '@testing-library/react';

import { conversations } from '../../../templates';
import List from '../index';

describe('components | List', () => {
  const defaultProps = {
    conversations,
    onCellClick: jest.fn(),
  };

  it('should match the snapshot', () => {
    const { asFragment } = render(<List {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should use onCellClick defaultProps', () => {
    const {
      defaultProps: { onCellClick },
    } = List;

    expect(onCellClick).toBeDefined();
    expect(onCellClick({ preventDefault: jest.fn() })).toBe(undefined);
  });
});
