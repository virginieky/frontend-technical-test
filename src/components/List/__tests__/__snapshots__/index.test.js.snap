// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match the snapshot 1`] = `
<DocumentFragment>
  <ul
    class="Wrapper-sc-1i3obe-0 bOCnmc"
  >
    <li
      class="Wrapper-sc-m0e7wi-0 gFNerB"
    >
      <div
        class="Padded-sc-q6j7xl-0 chxjCd"
      >
        <p
          class="Text-sc-yw3xjc-0 gQhyGc"
          color="black"
          font-size="16px"
          font-weight="regular"
        >
          6 6 Duarte - 1 Thibaut
        </p>
        <div
          class="Padded-sc-q6j7xl-0 dGTENR"
        />
        <p
          class="Text-sc-yw3xjc-0 fPFqtQ"
          color="black"
          font-size="12px"
          font-weight="regular"
        >
          Last message sent: 05/02/53693
        </p>
      </div>
    </li>
    <li
      class="Wrapper-sc-m0e7wi-0 gFNerB"
    >
      <div
        class="Padded-sc-q6j7xl-0 chxjCd"
      >
        <p
          class="Text-sc-yw3xjc-0 gQhyGc"
          color="black"
          font-size="16px"
          font-weight="regular"
        >
          5 5 Maeko - 1 Thibaut
        </p>
        <div
          class="Padded-sc-q6j7xl-0 dGTENR"
        />
        <p
          class="Text-sc-yw3xjc-0 fPFqtQ"
          color="black"
          font-size="12px"
          font-weight="regular"
        >
          Last message sent: 05/01/53693
        </p>
      </div>
    </li>
    <li
      class="Wrapper-sc-m0e7wi-0 gFNerB"
    >
      <div
        class="Padded-sc-q6j7xl-0 chxjCd"
      >
        <p
          class="Text-sc-yw3xjc-0 gQhyGc"
          color="black"
          font-size="16px"
          font-weight="regular"
        >
          4 5 Maeko - 1 Thibaut
        </p>
        <div
          class="Padded-sc-q6j7xl-0 dGTENR"
        />
        <p
          class="Text-sc-yw3xjc-0 fPFqtQ"
          color="black"
          font-size="12px"
          font-weight="regular"
        >
          Last message sent: 04/30/53693
        </p>
      </div>
    </li>
    <li
      class="Wrapper-sc-m0e7wi-0 gFNerB"
    >
      <div
        class="Padded-sc-q6j7xl-0 chxjCd"
      >
        <p
          class="Text-sc-yw3xjc-0 gQhyGc"
          color="black"
          font-size="16px"
          font-weight="regular"
        >
          3 1 Thibaut - 4 Elodie
        </p>
        <div
          class="Padded-sc-q6j7xl-0 dGTENR"
        />
        <p
          class="Text-sc-yw3xjc-0 fPFqtQ"
          color="black"
          font-size="12px"
          font-weight="regular"
        >
          Last message sent: 07/07/2021
        </p>
      </div>
    </li>
    <li
      class="Wrapper-sc-m0e7wi-0 gFNerB"
    >
      <div
        class="Padded-sc-q6j7xl-0 chxjCd"
      >
        <p
          class="Text-sc-yw3xjc-0 gQhyGc"
          color="black"
          font-size="16px"
          font-weight="regular"
        >
          1 2 Jeremie - 1 Thibaut
        </p>
        <div
          class="Padded-sc-q6j7xl-0 dGTENR"
        />
        <p
          class="Text-sc-yw3xjc-0 fPFqtQ"
          color="black"
          font-size="12px"
          font-weight="regular"
        >
          Last message sent: 07/07/2021
        </p>
      </div>
    </li>
    <li
      class="Wrapper-sc-m0e7wi-0 gFNerB"
    >
      <div
        class="Padded-sc-q6j7xl-0 chxjCd"
      >
        <p
          class="Text-sc-yw3xjc-0 gQhyGc"
          color="black"
          font-size="16px"
          font-weight="regular"
        >
          2 3 Patrick - 1 Thibaut
        </p>
        <div
          class="Padded-sc-q6j7xl-0 dGTENR"
        />
        <p
          class="Text-sc-yw3xjc-0 fPFqtQ"
          color="black"
          font-size="12px"
          font-weight="regular"
        >
          Last message sent: 05/06/2021
        </p>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
